Problems in the Javadoc of the JtxtUML API

During generating a Javadoc from the JtxtUML API and checking the result afterwards, I have discovered the following small problems:

- [x] In documentation comments of shorthand operations defined in ModelExecutor (`launch()`, `launch(Runnable)`, `run()`, `run(Runnable)`, `start(Runnable)`), the dots used for separating segments of the corresponding method chains are treated as sentence endings. As a result, in the generated method summary only the first segment of the chains is shown, which is really misleading. I believe this could be fixed by replacing these dots with their HTML entity number (`&#x2e;`).
- [x] `\api\model\execution\ModelExecutor.java:119: error: reference not found`
`"caused by them have been processed and all {@link TerminatePredicate}s"`
- [x] `\api\model\execution\ModelExecutor.java:134: error: reference not found`
`"{@link TerminationPredicate}s either)."`
- [x] `\api\model\execution\ModelExecutor.java:400: error: reference not found`
`"See {@link #setExecutorLog}."`
- [x] `\api\stdlib\timers\Timer.java:16: error: reference not found`
`"See the documentation of {@link hu.elte.txtuml.api.Model} for an overview on"`
- [ ] `\api\model\Runtime.java:111: warning - Tag @link:` 
`can't find connect(Class, Port, Class, Port) in hu.elte.txtuml.api.model.Action`
- [ ] `\api\model\Runtime.java:117: warning - Tag @link:`
`can't find connect(Port, Class, Port) in hu.elte.txtuml.api.model.Action`
- [ ] `\api\model\ModelClass.java:276: warning - Tag @see:`
`can't find connect(Port, Class, Port) in hu.elte.txtuml.api.model.Action`
- [ ] `\api\model\ModelClass.java:276: warning - Tag @see:`
`can't find connect(Class, Port, Class, Port) in hu.elte.txtuml.api.model.Action`

--- a/dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/ModelExecutor.java
+++ b/dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/ModelExecutor.java
@@ -81,7 +81,7 @@ public interface ModelExecutor extends BaseModelExecutor, Runnable {
 	ModelExecutor start() throws LockedModelExecutorException;
 
 	/**
-	 * A shorthand operation for {@link #setInitialization(Runnable)}.
+	 * A shorthand operation for {@link #setInitialization(Runnable)}&#x2e;
 	 * {@link #start()} (optional operation).
 	 * <p>
 	 * Supported iff {@link #setInitialization(Runnable)} is supported.
@@ -115,24 +115,26 @@ public interface ModelExecutor extends BaseModelExecutor, Runnable {
 
 	/**
 	 * Sets the model executor to be shut down after the currently running and
-	 * all scheduled actions have been performed and every non-external event
-	 * caused by them have been processed and all {@link TerminatePredicate}s
-	 * become true. To shut down the executor instantly, call
-	 * {@link #shutdownNow}.
+	 * all scheduled actions have been performed, every non-external event
+	 * caused by them have been processed and all <i>termination blockers</i> of
+	 * this model executor have been removed. To shut down the executor
+	 * instantly, call {@link #shutdownNow}.
 	 * <p>
 	 * This method <b>does not</b> await the termination of the executor, it
 	 * returns instantly.
 	 * 
 	 * @return this
 	 * @see #awaitTermination
+	 * @see #addTerminationBlocker(Object)
+	 * @see #removeTerminationBlocker(Object)
 	 */
 	ModelExecutor shutdown();
 
 	/**
 	 * Shuts down the model executor without waiting for any currently running
-	 * or scheduled actions to perform (it does not wait for
-	 * {@link TerminationPredicate}s either). In most cases, {@link #shutdown}
-	 * should be called instead.
+	 * or scheduled actions to perform (it does not wait for <i>termination
+	 * blockers</i> of this model executor to be removed either). In most cases,
+	 * {@link #shutdown} should be called instead.
 	 * <p>
 	 * This method <b>does not</b> await the termination of the executor, it
 	 * returns instantly.
@@ -181,7 +183,7 @@ public interface ModelExecutor extends BaseModelExecutor, Runnable {
 	void awaitTerminationNoCatch() throws InterruptedException;
 
 	/**
-	 * A shorthand operation for {@link #start()}.
+	 * A shorthand operation for {@link #start()}&#x2e;
 	 * {@link #awaitInitialization()}.
 	 * 
 	 * @return this
@@ -191,8 +193,9 @@ public interface ModelExecutor extends BaseModelExecutor, Runnable {
 	ModelExecutor launch() throws LockedModelExecutorException;
 
 	/**
-	 * A shorthand operation for {@link #setInitialization(Runnable)}.
-	 * {@link #start()}. {@link #awaitInitialization()} (optional operation).
+	 * A shorthand operation for {@link #setInitialization(Runnable)}&#x2e;
+	 * {@link #start()}&#x2e;{@link #awaitInitialization()} (optional
+	 * operation).
 	 * <p>
 	 * Supported iff {@link #setInitialization(Runnable)} is supported.
 	 * 
@@ -205,7 +208,7 @@ public interface ModelExecutor extends BaseModelExecutor, Runnable {
 	ModelExecutor launch(Runnable initialization) throws LockedModelExecutorException;
 
 	/**
-	 * A shorthand operation for {@link #start()}. {@link #shutdown()}.
+	 * A shorthand operation for {@link #start()}&#x2e;{@link #shutdown()}&#x2e;
 	 * {@link #awaitTermination()}.
 	 * 
 	 * @throws LockedModelExecutorException
@@ -215,9 +218,9 @@ public interface ModelExecutor extends BaseModelExecutor, Runnable {
 	void run() throws LockedModelExecutorException;
 
 	/**
-	 * A shorthand operation for {@link #setInitialization(Runnable)}.
-	 * {@link #start()}. {@link #shutdown()}. {@link #awaitTermination()}
-	 * (optional operation).
+	 * A shorthand operation for {@link #setInitialization(Runnable)}&#x2e;
+	 * {@link #start()}&#x2e;{@link #shutdown()}&#x2e;
+	 * {@link #awaitTermination()} (optional operation).
 	 * <p>
 	 * Supported iff {@link #setInitialization(Runnable)} is supported.
 	 * 
@@ -397,7 +400,7 @@ public interface ModelExecutor extends BaseModelExecutor, Runnable {
 	double getExecutionTimeMultiplier();
 
 	/**
-	 * See {@link #setExecutorLog}.
+	 * See {@link #setTraceLogging}.
 	 * 
 	 * @return whether execution log is switched on
 	 */

--- a/dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/base/AbstractModelExecutor.java
+++ b/dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/base/AbstractModelExecutor.java
@@ -346,8 +346,8 @@ public abstract class AbstractModelExecutor<S extends AbstractModelExecutor<S>> 
 	}
 
 	/**
-	 * Called if previously either {@link #shouldShutdownImmediately} or
-	 * {@link #shouldShutdownWhenNothingToDo} returned false and now it returns
+	 * Called if previously either {@link #shouldShutDownImmediately} or
+	 * {@link #shouldShutDownWhenNothingToDo} returned false and now it returns
 	 * true. All threads run by this model executor must be waken to check
 	 * whether they should shut down.
 	 * <p>

--- a/dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/base/ModelExecutorThread.java
+++ b/dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/base/ModelExecutorThread.java
@@ -10,14 +10,15 @@ import hu.elte.txtuml.utils.Logger;
 
 /**
  * A model executor thread is used by a model executor to run a model or parts
- * of a model (the executor may use multiple threads). Every {@link ModelClass}
- * and {@link Port} instance is owned by one model executor thread which runs
- * and manages that instance. Whenever an asynchronous event is raised (for
- * example, a signal is sent to such an instance), it is reported to the owner
- * thread, which puts that event into its mailbox and later processes it. The
- * processing of one such event is called a model execution step, which is also
- * an iteration of the main loop of a model executor thread. This thread stops
- * if:
+ * of a model (the executor may use multiple threads). Every
+ * {@link hu.elte.txtuml.api.model.ModelClass} and
+ * {@link hu.elte.txtuml.api.model.ModelClass.Port} instance is owned by one
+ * model executor thread which runs and manages that instance. Whenever an
+ * asynchronous event is raised (for example, a signal is sent to such an
+ * instance), it is reported to the owner thread, which puts that event into its
+ * mailbox and later processes it. The processing of one such event is called a
+ * model execution step, which is also an iteration of the main loop of a model
+ * executor thread. This thread stops if:
  * <ol>
  * <li>Either
  * <ul>
@@ -29,8 +30,8 @@ import hu.elte.txtuml.utils.Logger;
  * <li>or {@link #earlyStop} returns true (is false by default).</li>
  * </ul>
  * </li>
- * <li>The {@link ModelExecutor#unregisterThread(ModelExecutorThread)} method
- * returns true.</li>
+ * <li>The {@link AbstractModelExecutor#unregisterThread(ModelExecutorThread)}
+ * method returns true.</li>
  * </ol>
  * Otherwise, the loop tries to continue working, even if it has been
  * interrupted while waiting for new events.

--- a/dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/singlethread/SingleThreadModelClassWrapper.java
+++ b/dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/singlethread/SingleThreadModelClassWrapper.java
@@ -23,7 +23,7 @@ import hu.elte.txtuml.api.model.execution.impl.base.ModelExecutorThread;
 import hu.elte.txtuml.utils.InstanceCreator;
 
 /**
- * A {@link ModelClassWrapper} implementation for model executors that use only
+ * A {@link hu.elte.txtuml.api.model.runtime.ModelClassWrapper} implementation for model executors that use only
  * one model executor thread. This may not be used in a multi-thread executor as
  * it lacks the necessary synchronizations.
  */

--- a/dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/singlethread/SingleThreadModelExecutor.java
+++ b/dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/singlethread/SingleThreadModelExecutor.java
@@ -8,8 +8,9 @@ import hu.elte.txtuml.api.model.execution.impl.base.ModelExecutorThread;
 import hu.elte.txtuml.utils.Logger;
 
 /**
- * Abstract base class for {@link ModelExecutor} implementations which use only
- * one model executor thread.
+ * Abstract base class for
+ * {@link hu.elte.txtuml.api.model.execution.ModelExecutor} implementations
+ * which use only one model executor thread.
  */
 public abstract class SingleThreadModelExecutor<S extends SingleThreadModelExecutor<S>>
 		extends AbstractModelExecutor<S> {

--- a/dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/singlethread/SingleThreadPortWrapper.java
+++ b/dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/singlethread/SingleThreadPortWrapper.java
@@ -7,9 +7,9 @@ import hu.elte.txtuml.api.model.execution.impl.base.AbstractPortWrapper;
 import hu.elte.txtuml.api.model.execution.impl.base.AbstractSignalTargetWrapper;
 
 /**
- * A {@link PortWrapper} implementation for model executors that use only one
- * model executor thread. This may not be used in a multi-thread executor as it
- * lacks the necessary synchronizations.
+ * A {@link hu.elte.txtuml.api.model.runtime.PortWrapper} implementation for
+ * model executors that use only one model executor thread. This may not be used
+ * in a multi-thread executor as it lacks the necessary synchronizations.
  */
 public class SingleThreadPortWrapper extends AbstractPortWrapper {
 
@@ -19,12 +19,12 @@ public class SingleThreadPortWrapper extends AbstractPortWrapper {
 	public SingleThreadPortWrapper(Port<?, ?> wrapped, AbstractModelClassWrapper owner) {
 		super(wrapped, owner);
 	}
-	
+
 	@Override
 	public AbstractSignalTargetWrapper<?> getOuterConnection() {
 		return outer;
 	}
-	
+
 	@Override
 	public AbstractSignalTargetWrapper<?> getInnerConnection() {
 		return inner;

--- a/dev/plugins/hu.elte.txtuml.api.model/src/hu/elte/txtuml/api/model/Action.java
+++ b/dev/plugins/hu.elte.txtuml.api.model/src/hu/elte/txtuml/api/model/Action.java
@@ -223,7 +223,7 @@ public abstract class Action {
 	 * reception.
 	 * <p>
 	 * <b>Example:</b>
-	 * <p>
+	 * </p>
 	 * 
 	 * <pre>
 	 * <code>

--- a/dev/plugins/hu.elte.txtuml.api.model/src/hu/elte/txtuml/api/model/Model.java
+++ b/dev/plugins/hu.elte.txtuml.api.model/src/hu/elte/txtuml/api/model/Model.java
@@ -31,7 +31,7 @@ import hu.elte.txtuml.api.model.external.ExternalClass;
  * If a package is marked as a model then all {@code .java} files in that
  * package and its subpackages are considered parts of the model and therefore
  * should be valid model elements.
- * <p>
+ * </p>
  * 
  * <h1>Modeling in JtxtUML</h1>
  * 

--- a/dev/plugins/hu.elte.txtuml.api.stdlib/src/hu/elte/txtuml/api/stdlib/impl/timers/TimerImpl.java
+++ b/dev/plugins/hu.elte.txtuml.api.stdlib/src/hu/elte/txtuml/api/stdlib/impl/timers/TimerImpl.java
@@ -18,9 +18,9 @@ public class TimerImpl implements Timer {
 	/**
 	 * Creates a new {@code TimerImpl}.
 	 * 
-	 * @param obj
+	 * @param targetObj
 	 *            the target of the delayed send operation
-	 * @param s
+	 * @param signal
 	 *            the signal to send after the timeout
 	 * @param millisecs
 	 *            millisecs to wait before the timeout

--- a/dev/plugins/hu.elte.txtuml.api.stdlib/src/hu/elte/txtuml/api/stdlib/timers/Timer.java
+++ b/dev/plugins/hu.elte.txtuml.api.stdlib/src/hu/elte/txtuml/api/stdlib/timers/Timer.java
@@ -13,8 +13,8 @@ import hu.elte.txtuml.api.stdlib.impl.timers.TimerImpl;
  * send operation can be started, which means that a signal will be
  * asynchronously sent to the a target model object after a specified timeout.
  * <p>
- * See the documentation of {@link hu.elte.txtuml.api.Model} for an overview on
- * modeling in JtxtUML.
+ * See the documentation of {@link hu.elte.txtuml.api.model.Model} for an
+ * overview on modeling in JtxtUML.
  */
 public interface Timer extends ExternalClass {
 

